extension/surveys/survey-generator.js

```diff
@@ -276,13 +268,16 @@ ScaleQuestion.prototype.setAttributes = function(attributes) {
 ScaleQuestion.prototype.makeSingleRow =
     function(horizontal, questionName, reverse, showLabels) {
   var container = document.createElement('div');
-  var scaleElements = [];
-  for (var i = 0; i < this.scale.length; i++) {
+  var reversedScale = reverse ?
+      flipArray(
```

Comment:
I was looking at |flipArray|. It seems to duplicate the last item first by reversing the original array and then adding the last item again when |anchorLast| is true. Is that what you originally intended?